##### Exercise 3

###### a) Using the January 1976 - December 1979 four-year time period and the CAPM, generate variables measuring the GOLD-specific and the market risk premiums,and then estimate $\beta$ for GOLD. 

In [234]:
import numpy as np
import pandas as pd
from scipy import stats
import statsmodels.api as sm
from statsmodels.formula.api import ols

In [235]:
frame = pd.read_excel('chapter2data.xlsx', header=0, index_col='DATE')

In [236]:
gold   = frame.GOLD.loc  [pd.Timestamp('1976-01-01'):pd.Timestamp('1979-12-01')]
rkfree = frame.RKFREE.loc[pd.Timestamp('1976-01-01'):pd.Timestamp('1979-12-01')]
market = frame.MARKET.loc[pd.Timestamp('1976-01-01'):pd.Timestamp('1979-12-01')]

In [237]:
gold_specific       = gold - rkfree
market_risk_premium = market - rkfree

In [238]:
gold_slope, gold_intercept, gold_r_value, gold_p_value, gold_std_err = stats.linregress(gold_specific,
                                                                                        market_risk_premium)

In [239]:
print('alpha for GOLD:', gold_intercept)
print('betta for GOLD:', gold_slope)

alpha for GOLD: 0.0242045534692234
betta for GOLD: -0.23380356107998435


$\beta$ for GOLD is -0.2338

Compute 95% confidence interval for $\beta$

In [240]:
df_gold = pd.DataFrame({'x':market_risk_premium , 'y': gold_specific})
model_gold = ols("y ~ x", df_gold).fit()
model_gold.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.055
Model:                            OLS   Adj. R-squared:                  0.034
Method:                 Least Squares   F-statistic:                     2.672
Date:                Thu, 11 Oct 2018   Prob (F-statistic):              0.109
Time:                        23:26:34   Log-Likelihood:                 69.399
No. Observations:                  48   AIC:                            -134.8
Df Residuals:                      46   BIC:                            -131.1
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.0259      0.009      2.927      0.005       0.008       0.044
x             -0.2348      0.144     -1.635      0.109      -0.524       0.054
==============================================================================
Omnibus:                        4.217   Durbin-Watson:                   1.450
Prob(Omnibus):                  0.121   Jarque-Bera (JB):                3.071
Skew:                           0.505   Prob(JB):                        0.215
Kurtosis:                       3.718   Cond. No.                         17.1
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Confidence interval for $\beta$ is [-0.524,0.054] 

Why might such an asset be particularly desirable to an investor who is attempting to reduce risk through diversification? What does this imply concerning the expected return on such an asset?

###### b) Estimate the $\beta$ for GOLD using data from January 1980-December 1985 and construct a 95% confidence interval for $\beta$.

In [241]:
gold   = frame.GOLD.loc[pd.Timestamp('1980-01-01'):pd.Timestamp('1985-12-01')]
rkfree = frame.RKFREE.loc[pd.Timestamp('1980-01-01'):pd.Timestamp('1985-12-01')]
market = frame.MARKET.loc[pd.Timestamp('1980-01-01'):pd.Timestamp('1985-12-01')]

In [242]:
gold_specific = gold - rkfree
market_risk_premium = market - rkfree

In [243]:
gold_slope, gold_intercept, gold_r_value, gold_p_value, gold_std_err = stats.linregress(gold_specific,
                                                                                        market_risk_premium)
print('alpha for GOLD:', gold_intercept)
print('betta for GOLD:', gold_slope)

alpha for GOLD: 0.005724775562333638
betta for GOLD: 0.24836205036317202


$\beta$ for GOLD is 0.2484

In [247]:
df_gold = pd.DataFrame({'x':market_risk_premium , 'y': gold_specific})
model_gold = ols("y ~ x", df_gold).fit()
model_gold.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.101
Model:                            OLS   Adj. R-squared:                  0.088
Method:                 Least Squares   F-statistic:                     7.846
Date:                Thu, 11 Oct 2018   Prob (F-statistic):            0.00658
Time:                        23:28:27   Log-Likelihood:                 79.725
No. Observations:                  72   AIC:                            -155.4
Df Residuals:                      70   BIC:                            -150.9
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -0.0110      0.010     -1.152      0.253      -0.030       0.008
x              0.4058      0.145      2.801      0.007       0.117       0.695
==============================================================================
Omnibus:                       59.338   Durbin-Watson:                   1.683
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              369.487
Skew:                           2.362   Prob(JB):                     5.85e-81
Kurtosis:                      13.042   Cond. No.                         15.2
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Confidence interval for $\beta$ is [0.117,0.695] 

Comment on supply and demand shift factors possibly altering the $\beta$ of GOLD

##### Exercise 4

###### a) We need to show that using "the incorrect" reciprocal regression equation, one can still find the original CAPM parameters

In [251]:
Delta  = frame.DELTA.loc [pd.Timestamp('1983-01-01'):pd.Timestamp('1987-12-01')]
rkfree = frame.RKFREE.loc[pd.Timestamp('1983-01-01'):pd.Timestamp('1987-12-01')]
market = frame.MARKET.loc[pd.Timestamp('1983-01-01'):pd.Timestamp('1987-12-01')]

In [252]:
delta_risk_premium = Delta - rkfree
market_risk_premium = market - rkfree

In [255]:
delta_slope, delta_intercept, delta_r_value, delta_p_value, delta_std_err = stats.linregress( market_risk_premium,
                                                                                              delta_risk_premium)

In [259]:
print('inversed delta for DELTA:', delta_intercept)
print('inversed gamma for DELTA:', delta_slope)

inversed delta for DELTA: -0.004296310359829957
inversed gamma for DELTA: 0.6395249978002865


In [263]:
delta = -delta_intercept/delta_slope
print(delta)

0.006717970954392039


In [265]:
gamma = 1/delta_slope
print(gamma)

1.5636605346774641


In [266]:
delta_slope, delta_intercept, delta_r_value, delta_p_value, delta_std_err = stats.linregress(delta_risk_premium,
                                                                                             market_risk_premium)

In [268]:
print('alpha for DELTA:', delta_intercept)
print('betta for DELTA:', delta_slope)

alpha for DELTA: 0.0037012138384455635
betta for DELTA: 0.2569508396385637


In [269]:
df_gold = pd.DataFrame({'x':market_risk_premium , 'y':delta_risk_premium})
model_gold = ols("y ~ x", df_gold).fit()
model_gold.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.164
Model:                            OLS   Adj. R-squared:                  0.150
Method:                 Least Squares   F-statistic:                     11.41
Date:                Thu, 11 Oct 2018   Prob (F-statistic):            0.00131
Time:                        23:36:12   Log-Likelihood:                 62.531
No. Observations:                  60   AIC:                            -121.1
Df Residuals:                      58   BIC:                            -116.9
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -0.0043      0.011     -0.383      0.703      -0.027       0.018
x              0.6395      0.189      3.377      0.001       0.260       1.019
==============================================================================
Omnibus:                        0.111   Durbin-Watson:                   2.160
Prob(Omnibus):                  0.946   Jarque-Bera (JB):                0.003
Skew:                          -0.004   Prob(JB):                        0.999
Kurtosis:                       2.967   Cond. No.                         16.9
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

$R^2$ equals to 0.164

In [168]:
df_gold = pd.DataFrame({'y':market_risk_premium , 'x':delta_risk_premium})
model_gold = ols("y ~ x", df_gold).fit()
model_gold.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.164
Model:                            OLS   Adj. R-squared:                  0.150
Method:                 Least Squares   F-statistic:                     11.41
Date:                Thu, 11 Oct 2018   Prob (F-statistic):            0.00131
Time:                        21:06:51   Log-Likelihood:                 89.887
No. Observations:                  60   AIC:                            -175.8
Df Residuals:                      58   BIC:                            -171.6
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.0037      0.007      0.521      0.604      -0.011       0.018
x              0.2570      0.076      3.377      0.001       0.105       0.409
==============================================================================
Omnibus:                        9.431   Durbin-Watson:                   1.929
Prob(Omnibus):                  0.009   Jarque-Bera (JB):               12.306
Skew:                          -0.568   Prob(JB):                      0.00213
Kurtosis:                       4.905   Cond. No.                         10.7
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

$R^2$ also equals ro 0.164